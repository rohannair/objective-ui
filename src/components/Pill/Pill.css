@import '../../styles/mixins';
@define-extend pill {
  flex-flow: 0;
  font-size: 0.7rem;
  padding: 3px 5px;
  text-align: center;
  background-color: #ccc;
  font-weight: bold;
  text-transform: uppercase;
  border-radius: 3px;
  margin-right: 5px;
}

.success {
  @extend pill;
  @mixin success;
}

.info {
  @extend pill;
  @mixin info;
}

.warning {
  @extend pill;
  @mixin warning;
}

.danger {
  @extend pill;
  @mixin danger;
}

