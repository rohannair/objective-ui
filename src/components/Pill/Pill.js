import React, { PropTypes } from 'react';
import styles from './Pill.css';
import classNames from 'classnames/bind';

let cx = classNames.bind(styles);

const Pill = p => {
  const className = cx({
    [styles.info]: p.info,
    [styles.warning]: p.warning,
    [styles.danger]: p.danger,
    [styles.success]: p.success || (!p.info && !p.warning && !p.danger)
  });

  return (
    <span className={className}>
      { p.children }
    </span>
  );
};

export default Pill;
