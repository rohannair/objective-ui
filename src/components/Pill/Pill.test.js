import React from 'react'
import Pill from './index.js'

describe('Test suite for Pill component', () => {
  it('Pill should exist', () => {
    let wrapper = shallow(<Pill/>)
    expect(wrapper).toMatchSnapshot()
  })
})

