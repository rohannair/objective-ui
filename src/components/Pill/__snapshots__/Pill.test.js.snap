exports[`Test suite for Pill component Pill should exist 1`] = `
<span
  className="undefined" />
`;
